internal/subshell/sscommon/rcfile_test.go

```diff
@@ -43,35 +44,50 @@ func TestWriteRcFile(t *testing.T) {
 		path           string
 		env            map[string]string
 	}
+
+	zsh := fmt.Sprintf(
+		`export PATH="foo:$PATH"
+if [[ ! -z "$%s" && -f "$%s/%s" ]]; then
+  echo "State Tool is operating on project $%s, located at $%s"
+fi`,
+		constants.ActivatedStateEnvVarName,
+		constants.ActivatedStateEnvVarName,
+		constants.ConfigFileName,
+		constants.ActivatedStateNamespaceEnvVarName,
+		constants.ActivatedStateEnvVarName)
+	if runtime.GOOS == "windows" {
+		zsh = strings.ReplaceAll(zsh, "\n", "\r\n")
+	}
+
 	tests := []struct {
 		name         string
 		args         args
-		want error
+		want         error
 		wantContents string
 	}{
 		{
 			"Write RC to empty file",
 			args{
-				"fishrc_append.fish",
+				"zshrc_append.sh",
 				fakeFileWithContents("", "", ""),
 				map[string]string{
 					"PATH": "foo",
 				},
 			},
 			nil,
-			fakeContents("", `set -xg PATH "foo:$PATH"`, ""),
+			fakeContents("", zsh, ""),
 		},
 		{
 			"Write RC update",
 			args{
-				"fishrc_append.fish",
+				"zshrc_append.sh",
 				fakeFileWithContents("before", "SOMETHING ELSE", "after"),
 				map[string]string{
 					"PATH": "foo",
 				},
 			},
 			nil,
-			fakeContents(strings.Join([]string{"before", "after"}, fileutils.LineEnd), `set -xg PATH "foo:$PATH"`, ""),
+			fakeContents(strings.Join([]string{"before", "after"}, fileutils.LineEnd), zsh, ""),
```

Comment:
I don't think it makes sense to change the original tests here, they were testing something different, it seems. 

You could certainly add some tests here, but this feels like it's better covered with an integration test where we assert that the modifications were made rather than that they exactly match some expected output letter for letter, as that could become painful to maintain.

Reply:
I'm not sure I agree since we need to treat modifications to user RC files VERY carefully. The last thing we want to do is have a superficial check that verifies our boundaries were written between a line or two we expect to see, but we accidentally deleted a line or reformatted something that breaks the user's RC file. Doing absolute checks is safer, and is worth the maintenance I think.